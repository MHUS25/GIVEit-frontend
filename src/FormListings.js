import React, { Component } from "react";

class FormListings extends Component {
  constructor(props) {
    super(props);

    this.initialState = {
      title: '',
      description: '',
      start_date: '',
      end_date: '',
      listing_type: '',
      location: ''
    };

    this.state = this.initialState;
  }
  submitForm = (ev) => {
    if(ev && ev.preventDefault) { ev.preventDefault() };
    this.props.handleSubmit(this.state);
    this.setState(this.initialState);}

  handleFormSumbit(title, description, start_date, end_date, listing_type, location) {
   this.submitForm();
   let body = JSON.stringify({listing: {title: title, description: description, start_date: start_date, end_date: end_date, listing_type: listing_type, location: location}})
   fetch('https://giveit-backend.herokuapp.com/listings' , {
     method: 'POST',
     headers: {
       'Content-Type': 'application/json'
     },
     body: body,
   })
 }
  render() {
   let formFields = {}
   return(
     <form>
       <input placeholder="Enter title" id="title" ref={input => formFields.title = input} type="text"/><br/>
       <input placeholder="Enter description" id="description" ref={input => formFields.description = input} type="text"/><br/>
       <input placeholder="Enter start_date" id="start_date" ref={input => formFields.start_date = input} type="text"/><br/>
       <input placeholder="Enter end_date" id="end_date" ref={input => formFields.end_date = input} type="text"/><br/>
       <input placeholder="Enter listing_type" id="listing_type" ref={input => formFields.listing_type = input} type="text"/><br/>
       <input placeholder="Enter location" id="location" ref={input => formFields.location = input} type="text"/><br/>

       <button onClick={ (e) => {
         this.handleFormSumbit(formFields.title.value, formFields.description.value, formFields.start_date.value, formFields.end_date.value, formFields.listing_type.value, formFields.location.value)
       }}>Submit</button>
     </form>
   )
 }
}

export default FormListings;
