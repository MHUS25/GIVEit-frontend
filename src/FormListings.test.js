import React from 'react';
import FormListings from './components/SideBar'
import {shallow, mount} from 'enzyme'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('FormListings', () => {
  it('should render without crashing', () => {
    shallow(<FormListings/>);
  });
})

it('calls onClickMockHandler event on form submit', () => {
  const onClickMockHandler = jest.fn();
  const props = {
        handleSubmit: onClickMockHandler
    }
  let wrapper = mount(<FormListings {...props}/>);
  wrapper.find('#submit').simulate('click');
  expect(onClickMockHandler).toHaveBeenCalled();

});
