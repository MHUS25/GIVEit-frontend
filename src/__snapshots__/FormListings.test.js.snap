// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormListings should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormListings
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          Title
        </label>,
        <input
          id="title"
          name="title"
          onChange={[Function]}
          type="text"
          value=""
        />,
        <label>
          Description
        </label>,
        <input
          id="description"
          name="description"
          onChange={[Function]}
          type="text"
          value=""
        />,
        <label>
          Start Date
        </label>,
        <input
          id="start_date"
          name="start_date"
          onChange={[Function]}
          type="text"
          value=""
        />,
        <label>
          End Date
        </label>,
        <input
          id="end_date"
          name="end_date"
          onChange={[Function]}
          type="text"
          value=""
        />,
        <label>
          Listing Type
        </label>,
        <input
          id="listing_type"
          name="listing_type"
          onChange={[Function]}
          type="text"
          value=""
        />,
        <label>
          Location
        </label>,
        <input
          id="location"
          name="location"
          onChange={[Function]}
          type="text"
          value=""
        />,
        <input
          id="submit"
          onClick={[Function]}
          type="submit"
          value="submit"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title",
        },
        "ref": null,
        "rendered": "Title",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "title",
          "name": "title",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Description",
        },
        "ref": null,
        "rendered": "Description",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "description",
          "name": "description",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Start Date",
        },
        "ref": null,
        "rendered": "Start Date",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "start_date",
          "name": "start_date",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "End Date",
        },
        "ref": null,
        "rendered": "End Date",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "end_date",
          "name": "end_date",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Listing Type",
        },
        "ref": null,
        "rendered": "Listing Type",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "listing_type",
          "name": "listing_type",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Location",
        },
        "ref": null,
        "rendered": "Location",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "location",
          "name": "location",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "submit",
          "onClick": [Function],
          "type": "submit",
          "value": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Title
          </label>,
          <input
            id="title"
            name="title"
            onChange={[Function]}
            type="text"
            value=""
          />,
          <label>
            Description
          </label>,
          <input
            id="description"
            name="description"
            onChange={[Function]}
            type="text"
            value=""
          />,
          <label>
            Start Date
          </label>,
          <input
            id="start_date"
            name="start_date"
            onChange={[Function]}
            type="text"
            value=""
          />,
          <label>
            End Date
          </label>,
          <input
            id="end_date"
            name="end_date"
            onChange={[Function]}
            type="text"
            value=""
          />,
          <label>
            Listing Type
          </label>,
          <input
            id="listing_type"
            name="listing_type"
            onChange={[Function]}
            type="text"
            value=""
          />,
          <label>
            Location
          </label>,
          <input
            id="location"
            name="location"
            onChange={[Function]}
            type="text"
            value=""
          />,
          <input
            id="submit"
            onClick={[Function]}
            type="submit"
            value="submit"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title",
          },
          "ref": null,
          "rendered": "Title",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "title",
            "name": "title",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Description",
          },
          "ref": null,
          "rendered": "Description",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "description",
            "name": "description",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Start Date",
          },
          "ref": null,
          "rendered": "Start Date",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "start_date",
            "name": "start_date",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "End Date",
          },
          "ref": null,
          "rendered": "End Date",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "end_date",
            "name": "end_date",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Listing Type",
          },
          "ref": null,
          "rendered": "Listing Type",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "listing_type",
            "name": "listing_type",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Location",
          },
          "ref": null,
          "rendered": "Location",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "location",
            "name": "location",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "submit",
            "onClick": [Function],
            "type": "submit",
            "value": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
